### ISSUE TITLE
使用paddle进行简单图像分类，使用了随机种子仍然无法保证可复现性

### ISSUE THREAD

**Issue by thgpddl** [PaddlePaddle/Paddle#53104]
### bug描述 Describe the Bug

我使用aistudio进行简单的猫狗分类，项目已经公开：猫狗分类（Paddle2.0）_副本_副本：https://aistudio.baidu.com/aistudio/projectdetail/6003231?sUid=927072&shared=1&ts=1681924665793

程序是入口的jupyter文件，在开始导入库时我就设置了paddle和np的随机种子为120，但是在：
```python
model = paddle.Model(MyCNN())
model.prepare(paddle.optimizer.Adam(learning_rate=0.0005, parameters=model.parameters()), 
              paddle.nn.CrossEntropyLoss(), 
              paddle.metric.Accuracy())
# 训练可视化VisualDL工具的回调函数
visualdl = paddle.callbacks.VisualDL(log_dir='visualdl_log')
# 启动模型全流程训练

model.fit(train_dataset, epochs=10, batch_size=256, verbose=1)
#保存模型
model.save('model_save_dir')
```
该部分训练时，多次启动其结果不一致，我附图三次训练的记录，其每次精度总是不一致
![image](https://user-images.githubusercontent.com/48787805/233151270-912bfccf-620b-4462-8e01-291d66e9a818.png)
![image](https://user-images.githubusercontent.com/48787805/233151269-4c7cdd8f-dcf0-4d82-8108-95f45b0ec852.png)
![image](https://user-images.githubusercontent.com/48787805/233151271-2b51de89-31d5-4608-bbe3-5d234c3c9345.png)

我的问题是，我如何设置才能够保证每次训练的指标变化是完全一致的

**Comment by lym0302** [comment]
您好，可以自行排查下每次模型的输入是否一样

**Comment by thgpddl** [comment]
> 您好，可以自行排查下每次模型的输入是否一样

输入是一样的

**Comment by thgpddl** [comment]
我进行了一些实验：
1）在aistudio平台设置paddle.seed(0)随机种子，使用CPU训练，能够对齐
2）在其他云GPU使用3090，设置paddle.seed(0)随机种子，能够对齐
3）在aistudio平台设置paddle.seed(0)随机种子，使用GPU训练，不能够对齐
综上，至少我认为aistudio+GPU+paddle是无法通过设置随机种子对齐的，是否存在我不了解的其他设置目前未可知，还没有任何官方的有效回复。

**Comment by thgpddl** [comment]
我进行了一些实验：
1）在aistudio平台设置paddle.seed(0)随机种子，使用CPU训练，能够对齐
2）在其他云GPU使用3090，设置paddle.seed(0)随机种子，~~能够对齐~~（对齐个屁，之前是epoch太小，设置大一点epoch还是对不齐）
3）在aistudio平台设置paddle.seed(0)随机种子，使用GPU训练，不能够对齐

只有在cpu上能对齐，在gpu上就无法对齐

**Comment by Ligoml** [comment]
可以试一下这样能不能固定gpu训练中的随机种子：https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/flags/cudnn_cn.html#flags-cudnn-deterministic

**Comment by lym0302** [comment]
你好，export FLAGS_cudnn_deterministic=True 可以使得每次结果一致
